mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -1061,6 +1066,67 @@ class ConstructorPattern final
   }
 };
 
+//===----------------------------------------------------------------------===//
+// AccessorGetPointerPattern - Convert `sycl.accessor.get_pointer` to LLVM.
+//===----------------------------------------------------------------------===//
+
+class AccessorGetPointerPattern
+    : public ConvertOpToLLVMPattern<SYCLAccessorGetPointerOp>,
+      public GetMemberPattern<AccessorGetPtr>,
+      public GetMemberPattern<AccessorGetID, IDGetDim>,
+      public GetMemberPattern<AccessorGetMemRange, RangeGetDim> {
+public:
+  using ConvertOpToLLVMPattern<
+      SYCLAccessorGetPointerOp>::ConvertOpToLLVMPattern;
+
+private:
+  template <typename... Args> Value getID(Args &&...args) const {
+    return GetMemberPattern<AccessorGetID, IDGetDim>::loadValue(
+        std::forward<Args>(args)...);
+  }
+  template <typename... Args> Value getMemRange(Args &&...args) const {
+    return GetMemberPattern<AccessorGetMemRange, RangeGetDim>::loadValue(
+        std::forward<Args>(args)...);
+  }
+
+  Value getTotalOffset(OpBuilder &builder, Location loc, AccessorType accTy,
+                       OpAdaptor opAdaptor) const {
+    const auto acc = opAdaptor.getAcc();
+    const auto resTy = builder.getI64Type();
+    Value res = builder.create<arith::ConstantIntOp>(loc, 0, resTy);
+    for (unsigned i = 0, dim = accTy.getDimension(); i < dim; ++i) {
+      // Res = Res * Mem[I] + Id[I]
+      const auto memI = getMemRange(builder, loc, resTy, acc, i);
+      const auto idI = getID(builder, loc, resTy, acc, i);
+      res = builder.create<arith::AddIOp>(
+          loc, builder.create<arith::MulIOp>(loc, res, memI), idI);
+    }
+    return res;
+  }
+
+public:
+  LogicalResult
+  matchAndRewrite(SYCLAccessorGetPointerOp op, OpAdaptor opAdaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    const auto loc = op.getLoc();
+    const Value zero = rewriter.create<arith::ConstantIntOp>(loc, 0, 64);
```

Comment:
[nit]: const Value zero -> const auto zero. 
Also the length of the constant is hard coded to 64. Instead of hard coding to an i64 should we use `builder.getIndexType()` here and other places where we get the offset?